manager/integration/tests/test_node.py

```diff
@@ -2184,3 +2184,50 @@ def test_replica_scheduler_rebuild_restore_is_too_big(client):  # NOQA
     node = common.wait_for_disk_update(client, lht_hostId,
                                        len(update_disks))
     cleanup_host_disk(client, 'vol-small')
+
+
+@pytest.mark.skip(reason="TODO")
+def test_node_eviction():
+    """
+    Test node eviction (assuming this is a 3 nodes cluster)
+
+    1. Disable scheduling on node 3.
+    2. Create volume 1 with 2 replicas.
+    3. Attach volume 1 to node 1 and write some data to it and get
+    checksum 1.
+    4. Disable scheduling and set 'Eviction Requested' to 'true' on node 1.
+    5. Volume 1 should be failed to schedule new replica.
+    6. Set 'Eviction Requested' to 'false' to cancel node 1 eviction and
+    check there should be 1 replica on node 1 and node 2. And remove the
+    unscheduled replica.
+    7. Set 'Eviction Requested' to 'true' on node 1.
+    8. Set 'Replica Node Level Soft Anti-Affinity' to 'true'.
+    9. The eviction should be success, and no replica on node 1, 2 replicas
+    on node 2.
+    10. Enable scheduling on node 3, and set 'Eviction Requested' to
+    'false', enable scheduling on node 1.
+    11. Set 'Replica Node Level Soft Anti-Affinity' to 'false'.
+    12. Disable scheduling and set 'Eviction Requested' to 'true' on node 2.
+    13. The eviction should be success and no replica on node 2. And 1
```

Comment:
Verify the volume keeps `Healthy` during the eviction

Reply:
Do you mean to check the volume state after each step?

Reply:
No. I just want to verify the volume HA is not affected during the eviction. Hence checking the volume robustness during the eviction is enough.

Reply:
Sure, I just updated step 12 to make sure volume is in healthy state during the eviction.